libhwmon/wrapadl.cpp

```diff
@@ -180,7 +245,8 @@ int wrap_adl_get_tempC(wrap_adl_handle *adlh, int gpuindex, unsigned int *tempC)
 		return -1;
 
 	ADLTemperature *temperature = new ADLTemperature();
-	rc = adlh->adlOverdrive5TemperatureGet(adlh->phys_logi_device_id[gpuindex], 0, temperature);
+	int adlidx = adlh->opencl_adl_device_id[gpuindex];
```

Comment:
gpuindex should be always adl id, the caller should use the mapping array if needed